components/src/dynamo/planner/tests/unit/test_prometheus_traffic_provider.py

```
@@ -133,3 +136,168 @@ async def test_invalid_required_metric_is_checked_before_float_logging(missing_m
+    client.get_avg_spec_decode_accept_length.return_value = 2.5
+    try:
+        yield provider, client
+    finally:
+        if provider._collection_task is not None:
+            await asyncio.gather(provider._collection_task, return_exceptions=True)
+        client_patch.stop()
+
+
+@pytest.fixture
+async def blocked_query(traffic_provider):
+    provider, client = traffic_provider
```

Ruff (0.16.3)

[warning] Unpacked variable `provider` is never used

(RUF059)